pkg/cable/libreswan/libreswan.go

```diff
@@ -498,13 +497,14 @@ func removeConnectionStanza(confPath, connName string) error {
 func (i *libreswan) connectToEndpointCertMode(endpointInfo *natdiscovery.NATEndpointInfo) (string, error) {
 	logger.Info("Certificate mode: skipping pluto start and whack; assuming pluto is managed externally and config/NSS DB are updated.")
 
-	confPath := submarinerConfPath
+	confPath := SubmarinerConfPath()
 	endpoint := &endpointInfo.Endpoint
 	leftID := "submariner-client-" + i.localEndpoint.ClusterID
 	left := i.localEndpoint.GetPrivateIP(endpointInfo.UseFamily)
 	right := endpointInfo.UseIP
-	leftSubnets := i.localEndpoint.Subnets
-	rightSubnets := endpoint.Spec.Subnets
+
+	leftSubnets := i.localEndpoint.ExtractSubnetsExcludingIP(endpointInfo.UseIP)
+	rightSubnets := endpoint.Spec.ExtractSubnetsExcludingIP(endpointInfo.UseIP)
```

Comment:
@aswinsuryan  Elsewhere it uses `ExtractSubnetsExcludingIP` so I assume we shoild do the same here?

Reply:
+1 , this may be required , though the default use case works